# Thursday March 27th

## Exercise One

### Read titanic.csv file using Pandas, load into vscode as a dataframe, save into a dataframe

In [38]:
import numpy as np
import pandas as pd

In [39]:
#Exercise 1

df = pd.read_csv(r"/Users/Toki3Duckie/code/SavvyCoders/DAP-Curriculum/Section05/resources/titanic.csv")
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

# Exercise 2
## There are 891 records of passengers aborad the Titanic. How many survived? 
## Bonus 1: How many men survived versus how many women survived

In [40]:
#Hint - df.value_counts()

#print(df.value_counts("Survived"))
# Hint#2

#How many did not survive?
#within brackets the column name, the zero is for 0 as in died value in the column, 2 for the index of the column
print(df.value_counts("Survived").get(0, 2))

#full data for survive and did not survive
df.value_counts("Survived")

549


Survived
0    549
1    342
Name: count, dtype: int64

In [41]:
#Bonus

print(df.value_counts(subset = ["Sex", "Survived"]))

Sex     Survived
male    0           468
female  1           233
male    1           109
female  0            81
Name: count, dtype: int64


In [42]:
#AlternativeVersion for getting counts for columns

df["Survived"].value_counts()

Survived
0    549
1    342
Name: count, dtype: int64

# Exercise Three

## Sort the passengers by ticket price. What was the highest paid ticket? What were the lowest three?

df.sort_values(by = ["Fare"], inplace= True, ascending = False)

In [ ]:
#Exercise 3 Bonus without resorting the dataframe find the highest and lowest

print(df["Fare"].max())

print(df["Fare"].min())

#print(range_df["Fare"].min())

512.3292
0.0


NameError: name 'range_df' is not defined

In [ ]:
#Exercise 3

df.sort_values(by = ["Fare"], inplace= True, ascending = True)
df
print(df.head((4)))
print(df.tail(1))



     PassengerId  Survived  Pclass                              Name   Sex  \
815          816         0       1                  Fry, Mr. Richard  male   
806          807         0       1            Andrews, Mr. Thomas Jr  male   
413          414         0       2    Cunningham, Mr. Alfred Fleming  male   
263          264         0       1             Harrison, Mr. William  male   
633          634         0       1     Parr, Mr. William Henry Marsh  male   
732          733         0       2              Knight, Mr. Robert J  male   
674          675         0       2        Watson, Mr. Ennis Hastings  male   
179          180         0       3               Leonard, Mr. Lionel  male   
277          278         0       2       Parkes, Mr. Francis "Frank"  male   
466          467         0       2             Campbell, Mr. William  male   
822          823         0       1   Reuchlin, Jonkheer. John George  male   
597          598         0       3               Johnson, Mr. Al

# Exercise Four

## There are three classes of passenger, first,second, and third (1,2, and 3). Calculate the survival rate of each

In [61]:
##Exercise 4 Bonus

#personal question about the ratio of mortality by gender
#81/233
#109/468

print(df["Pclass"].value_counts())
print(df.value_counts(subset = ["Survived", "Pclass"]))

survival_firstclass = (80/216)*100
survival_secondclass = (97/184)*100
survival_thirdclass = (372/491)*100

print(f"{survival_firstclass} percent of first class folks died on the titanic")
print(f"{survival_secondclass} percent of second class folks died")
print(f"{survival_thirdclass} percent of first class folks died")


Pclass
3    491
1    216
2    184
Name: count, dtype: int64
Survived  Pclass
0         3         372
1         1         136
          3         119
0         2          97
1         2          87
0         1          80
Name: count, dtype: int64
37.03703703703704 percent of first class folks died on the titanic
52.71739130434783 percent of second class folks died
75.76374745417516 percent of first class folks died
